Cython/Compiler/Optimize.py

```diff
@@ -240,6 +240,8 @@ def _optimise_for_loop(self, node, iterable, reversed=False):
         # dict iteration?
         if function.is_attribute and not reversed and not arg_count:
             base_obj = iterable.self or function.obj
+            if isinstance(base_obj, ExprNodes.ProxyNode):
+                base_obj = base_obj.arg
```

Comment:
Special cases like this always make me wonder if they're really special cases or whether we're just finding them because there happens to be a test that triggers them (and for the "more general case" that also fails, there happens to be no such test). :-/

Reply:
Indeed - that's what I think I meant by "...but does feel a little fragile".